R/functions_util.R

```diff
@@ -40,24 +40,57 @@ FormatString = function(x, quote=TRUE, sep=", ") {
   return(glue::glue(x, .transformer=GlueTransformer_quote_collapse(), .envir=parent.frame()))
 }
 
+#' Generates a callout box for showing notes, tips or warnings
+#' 
+#' @param x Character string that will be formatted by the FormatString function
+#' @param type Type of callout box. Can be: 'note', 'tip', 'important', 'caution' and 'warning'.
+#' @param print Whether to print (if TRUE) or to return (FALSE) the message box
+#' @return Character string to generate a message box
+CalloutBox = function(x, type, print=TRUE) {
+  valid_types = c("note", "tip", "important", "caution", "warning")
+  assertthat::assert_that(type %in% valid_types,
+                          msg=FormatString("Callout box typ {type} but must be one of: {valid_types*}."))
+  
+  
+  
+  x = paste0("\n\n::: callout-", type, "\n", FormatString(x), "\n:::\n\n")
+  if (print) {
+    cat(x)
+  } else {
+    return(x)
+  }
+}
+
 #' Generates a warning box
 #' 
 #' @param x Character string that will be formatted by the FormatString function
+#' @param print Whether to print (if TRUE) or to return (FALSE) the warning box
 #' @return Character string to generate a warning box
-WarningBox = function(x) {
-  x = paste0("\n::: callout-warning\n", FormatString(x), "\n:::\n")
-  cat(x)
+WarningBox = function(x, print=TRUE) {
```

Comment:
This is doubled. Let's just keep the new general `CalloutBox` function, and remove these.